File: September-2025/13_09_2025_gfg.py
def minimumCostToCutBoard(n : int, m : int, x : list[int], y : list[int]) -> int:
    """
    Given a board of dimensions n x m that needs to be cut into n*m squares.
    The cost of making a cut along a horizontal or vertical edge is provided in two arrays:

    x[]: Cutting costs along the vertical edges (length-wise).
    y[]: Cutting costs along the horizontal edges (width-wise).

    Find the minimum total cost required to cut the board into squares optimally.
    """
    x.sort(reverse=True)
    y.sort(reverse=True)
    
    horizontal_segments = 1
    vertical_segments = 1
    
    x_index = 0
    y_index = 0
    
    total_cost = 0
    
    while x_index < len(x) and y_index < len(y):
        if x[x_index] > y[y_index]:
            total_cost += x[x_index] * horizontal_segments
            vertical_segments += 1
            x_index += 1
        else:
            total_cost += y[y_index] * vertical_segments
            horizontal_segments += 1
            y_index += 1
            
    while x_index < len(x):
        total_cost += x[x_index] * horizontal_segments
        vertical_segments += 1
        x_index += 1
        
    while y_index < len(y):
        total_cost += y[y_index] * vertical_segments
        horizontal_segments += 1
        y_index += 1
        
    return total_cost

File: September-2025/13_09_2025_gfg.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
  public:
    int minCost(int n, int m, vector<int>& x, vector<int>& y) {
        sort(x.begin(), x.end(), greater<int>());
        sort(y.begin(), y.end(), greater<int>());

        int horizontal_segments = 1;
        int vertical_segments = 1;
        int x_index = 0;
        int y_index = 0;
        int total_cost = 0;

        while (x_index < x.size() && y_index < y.size()) {
            if (x[x_index] > y[y_index]) {
                total_cost += x[x_index] * horizontal_segments;
                vertical_segments++;
                x_index++;
            } else {
                total_cost += y[y_index] * vertical_segments;
                horizontal_segments++;
                y_index++;
            }
        }

        while (x_index < x.size()) {
            total_cost += x[x_index] * horizontal_segments;
            vertical_segments++;
            x_index++;
        }

        while (y_index < y.size()) {
            total_cost += y[y_index] * vertical_segments;
            horizontal_segments++;
            y_index++;
        }

        return total_cost;
    }
};